### PR TITLE
=www-client/seamonkey-2.53.18.2-r1: Various improvements

### PR DESCRIPTION
**Pull request by BioMike**
* Patch for fixing build with rust-1.78.0 and newer.
* Bumped llvm/clang support to 17 and 18.

Closes: https://bugs.gentoo.org/935415

---

Please check all the boxes that apply:

- [ x ] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [ x ] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [ x ] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [ x ] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: www-client/seamonkey/seamonkey-2.53.18.2-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+
+SRC_URI="https://archive.seamonkey-project.org/releases/${MOZ_PV}/source/${MY_MOZ_P}.source.tar.xz -> ${P}.source.tar.xz
+	https://archive.seamonkey-project.org/releases/${MOZ_PV}/source/${MY_MOZ_P}.source-l10n.tar.xz -> ${P}.source-l10n.tar.xz
+	https://github.com/BioMike/gentoo-${PN}-patches/archive/refs/tags/${PATCH_PV}.tar.gz -> ${PATCH}.tar.gz"
+
+S="${WORKDIR}/${MY_MOZ_P}"
+
+MOZ_GENERATE_LANGPACKS=1
+MOZ_L10N_SOURCEDIR="${S}/${P}-l10n"
+inherit autotools check-reqs desktop edos2unix flag-o-matic llvm mozcoreconf-v6 mozlinguas-v2 pax-utils \
```
Comment (juippis) [inline www-client/seamonkey/seamonkey-2.53.18.2-r1.ebuild:36]: Could you migrate to *llvm-r1.eclass*?

File: www-client/seamonkey/seamonkey-2.53.18.2-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,568 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+WANT_AUTOCONF="2.1"
+
+PYTHON_COMPAT=( python3_{10..11} )
```
Comment (juippis) [inline www-client/seamonkey/seamonkey-2.53.18.2-r1.ebuild:8]: ... and no newer python support? :)

**Review by juippis — APPROVED** [review]
It works, would appreciate if you migrated to *llvm-r1* eclass and figured if there's anewer python support. Will merge a bit later.

File: www-client/seamonkey/seamonkey-2.53.18.2-r1.ebuild (hunk 2 above)
Comment (BioMike) [reply www-client/seamonkey/seamonkey-2.53.18.2-r1.ebuild:8]: No, python 3.12 isn't supported yet, build system calls deprecated functions, which are removed in 3.12. I'm working with upstream to get this in.

File: www-client/seamonkey/seamonkey-2.53.18.2-r1.ebuild (hunk 1 above)
Comment (BioMike) [reply www-client/seamonkey/seamonkey-2.53.18.2-r1.ebuild]: I've just committed this migration, I hope I did it correctly (a test build did work).

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**